src/Database/Query.php

```diff
@@ -1412,11 +1413,16 @@ public function values($data)
      *
      * Can be combined with set() and where() methods to create update queries.
      *
-     * @param string $table The table you want to update.
+     * @param string|\Cake\Database\Expression\QueryExpression $table The table you want to update.
      * @return $this
      */
     public function update($table)
     {
+        if (!is_string($table) && !($table instanceof QueryExpression)) {
+            $message = sprintf('Table must be of type string, got "%s"', gettype($table));
```

Comment:
For the sake of completeness, something along the lines of

> or an instance of `` `\Cake\Database\ExpressionInterface` ``

should probably be included too.